### PR TITLE
Flex reverse sequence op

### PR DESCRIPTION
**Pull request by XingHongChenIntel**
new version sequence op, replaced slice op by gather op.

### CODE REVIEW

File: plaidml/bridge/openvino/ops/reverse_sequence.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  } else {
+    THROW_IE_EXCEPTION << " input [1] is Unsupported inputType; ";
+  }
+}
+
+edsl::Tensor reverse_tensor(edsl::Tensor reverse_crop, int64_t seq_axis) {
+  std::vector<edsl::TensorDim> dims(reverse_crop.rank());
+  reverse_crop.bind_dims(dims);
+  std::vector<edsl::TensorIndex> I_idxs(reverse_crop.rank());
+  std::vector<edsl::TensorIndex> O_idxs;
```
Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/reverse_sequence.cpp:30]: can we use TensorIndex O_idxs[some_size] here? we should avoid using C++ if it is possible.

Comment (YangleiZouIntel) [reply plaidml/bridge/openvino/ops/reverse_sequence.cpp:30]: This still using c++ array but it is more close to edsl code style.

Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/reverse_sequence.cpp]: I left a comment and deleted here. I think we can leave it as it is. we are using std::vector in many other ops anyway.

File: plaidml/bridge/openvino/ops/reverse_sequence.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  for (int64_t axis = 0; axis < reverse_crop.rank(); axis++) {
+    if (axis == seq_axis) {
+      O_idxs.push_back(dims[axis] - 1 - I_idxs[axis]);
+    } else {
+      O_idxs.push_back(I_idxs[axis]);
+    }
```
Comment (YangleiZouIntel) [inline plaidml/bridge/openvino/ops/reverse_sequence.cpp]: Can you use something like:
```
std::vector<edsl::TensorIndex> O_idxs=I_idxs;
O_idxs[seq_axis]=dims[seq_axis] - 1 - I_idxs[seq_axis];
```

Comment (YangleiZouIntel) [reply plaidml/bridge/openvino/ops/reverse_sequence.cpp]: Is this the same as the original code？

Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/ops/reverse_sequence.cpp]: ok, I will try it .

Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/ops/reverse_sequence.cpp]: yes, you are right, it's totally same as the original code, I fixed it.

**Comment by YangleiZouIntel** [comment]
LGTM, please submit  this to plaidml/plaidml

**Comment by YangleiZouIntel** [comment]
submitted to plaidml

**Pull request closed without merging**